storagemarket/deal_handler.go

```diff
@@ -31,14 +31,74 @@ type dealHandler struct {
 	transferMu       sync.Mutex
 	transferFinished bool
 	transferErr      error
+
+	updateSubsLk sync.RWMutex
+	updateSubs   map[*updatesSubscription]struct{}
+}
+
+func newDealHandler(ctx context.Context, dealUuid uuid.UUID) *dealHandler {
+	// Create a deal handler
+	bus := eventbus.NewBus()
+
+	transferCtx, cancel := context.WithCancel(ctx)
+	return &dealHandler{
+		providerCtx: ctx,
+		dealUuid:    dealUuid,
+		bus:         bus,
+
+		transferCtx:    transferCtx,
+		transferCancel: cancel,
+		transferDone:   make(chan error, 1),
+
+		updateSubs: make(map[*updatesSubscription]struct{}),
+	}
+}
+
+// updatesSubscription wraps event.Subscription so that we can add an onClose
+// callback
+type updatesSubscription struct {
+	event.Subscription
+	onClose func(*updatesSubscription)
 }
 
+func (s *updatesSubscription) Close() error {
```

Comment:
Where do we call close on the subscription ?

Reply:
Yes, it's called when the front end stops subscribing to events.
This can happen because
- the user closes the web page
- a new deal comes in that moves the last deal on the page to page 2 (ie no longer visible)